mlx/backend/metal/kernels/quantized.h

```diff
@@ -628,10 +628,12 @@ METAL_FUNC void qmv_impl(
       const device T* sl = scales + row * in_vec_size_g;
       const device T* bl = biases + row * in_vec_size_g;
 
-      U s = sl[0];
-      U b = bl[0];
-      result[row] += qdot_safe<U, values_per_thread, bits>(
-          wl, x_thread, s, b, sum, remaining);
+      if (remaining > 0) {
+        U s = sl[0];
+        U b = bl[0];
+        result[row] += qdot_safe<U, values_per_thread, bits>(
+            wl, x_thread, s, b, sum, remaining);
+      }
```

Comment:
Even though `sum` and `accum` are zero when `remaining=0`, `scale * accum + sum * bias can still produce bad outputs if the uninitialized memory at `s` or `b` is nan or inf

Reply:
That's awesome! Thanks for finding and fixing this! I can't be sure without measuring but a possibly better fix is to skip everything if remaining is 0. And it would also be needed at lines 563.